python/cuml/feature_extraction/text.py

```diff
@@ -0,0 +1,603 @@
+
+        Prune features that are non zero in more samples than high or less
+        documents than low, modifying the vocabulary, and restricting it to
+        at most the limit most frequent.
+        """
+        if high is None and low is None and limit is None:
+            self.stop_words_ = None
+            return count_df
+
+        document_frequency = _document_frequency(count_df)
+
+        mask = cp.ones(len(document_frequency), dtype=bool)
+        if high is not None:
+            mask &= document_frequency <= high
+        if low is not None:
+            mask &= document_frequency >= low
+        if limit is not None and mask.sum() > limit:
+            term_frequency = _term_frequency(count_df)
+            mask_inds = (-term_frequency[mask]).argsort()[:limit]
+            new_mask = cp.zeros(len(document_frequency), dtype=bool)
+            new_mask[cp.where(mask)[0][mask_inds]] = True
+            mask = new_mask
+
+        keep_idx = cp.where(mask)[0].astype(cp.int32)
+        keep_num = keep_idx.shape[0]
+
+        if keep_num == 0:
+            raise ValueError("After pruning, no terms remain. Try a lower"
+                             " min_df or a higher max_df.")
+
+        if len(vocab) - keep_num != 0:
+            count_df = self._filter_and_renumber(count_df, keep_idx, 'token')
+
+        self.stop_words_ = vocab[~mask].reset_index(drop=True)
+        self.vocabulary_ = vocab[mask].reset_index(drop=True)
+
+        return count_df
+
+    def _preprocess(self, raw_documents):
+        preprocess = self.build_preprocessor()
+        return preprocess(raw_documents)
+
+    def fit(self, raw_documents):
+        """Build a vocabulary of all tokens in the raw documents.
+
+       Parameters
+       ----------
+       raw_documents : cudf.Series
+           A Series of string documents
+
+       Returns
+       -------
+       self
+       """
+        self.fit_transform(raw_documents)
+        return self
+
+    def fit_transform(self, raw_documents):
+        """Build the vocabulary and return document-term matrix.
+
+        Equivalent to .fit(X).transform(X) but preprocess X only once.
+
+        Parameters
+        ----------
+        raw_documents : cudf.Series
+           A Series of string documents
+
+        Returns
+        -------
+        X : array of shape (n_samples, n_features)
+            Document-term matrix.
+        """
+        self._warn_for_unused_params()
+        self._validate_params()
+
+        self._fixed_vocabulary = self.vocabulary is not None
+
+        docs = self._preprocess(raw_documents)
+        n_doc = len(docs)
+
+        tokenized_df = self._create_tokenized_df(docs)
+
+        if self._fixed_vocabulary:
+            self.vocabulary_ = self.vocabulary
+        else:
+            self.vocabulary_ = tokenized_df["token"].unique()
+
+        count_df = self._count_vocab(tokenized_df)
+
+        if not self._fixed_vocabulary:
+            max_doc_count = (self.max_df
```

Comment:
oh, man, sometimes sklearn APIs are a little too flexible... your implementation seems good. Except maybe cast to int like `int(self.max_df * n_doc)`

Reply:
What do you mean? Replacing the whole statement with 
```python
max_doc_count = (self.max_df if isinstance(self.max_df, numbers.Integral) else int(self.max_df * n_doc))
```
Would not work because when `self.max_df` is not an interger, we expect `max_doc_count` to be float. For instance, if we set `max_df` to 0.5, meaning we filter out terms which appear in more than 50% of documents, and we have 3 documents, `max_doc_count` should be `1.5`. In this setting, a term appearing in 2 documents would be filtered out (term_count>50% or 2>1.5) and a term appearing in 1 would be kept (1<1.5).